### PR TITLE
Button refactor

### PR DESCRIPTION
**Pull request by bobwise**
## What does this PR do?
Updating all styles for buttons. We are moving away from red as our primary color.

## Please check off completed items as you work.
If a checklist item or section does not apply to your PR
then please remove it.
### Documentation
 - [x] Update Spark Docs Vanilla
 - [x] Update Spark Docs Angular
 - [x] Update Component Sass Var/Class Modifier table

### Code
 - [x] Build Component in Spark Vanilla (Sass, HTML, JS)
 - [x] Build Component in Spark Angular
 - [x] Unit Testing in Spark Vanilla (100% coverage, 100% passing)
 - [x] Unit Testing in Spark Angular (100% coverage, 100% passing)

### Accessibility
- [x] New changes abide by [accessibility requirements](https://sparkdesignsystem.com/docs/accessibility)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)

### Design Review
 - [x] Design reviewed and approved

Issue https://github.com/sparkdesignsystem/spark-design-system/issues/916

### CODE REVIEW

**Comment by bobwise** [comment]
design review in progress

File: packages/spark-core/components/_buttons.scss [hunk 1]
```diff
@@ -109,3 +109,21 @@
     color: $btn-disabled-text-color;
   }
 }
+
+.sprk-c-Button--secondary.sprk-is-Disabled,
+.sprk-c-Button--secondary[disabled] {
+  box-shadow: none;
+
```
Comment (afebbraro) [inline packages/spark-core/components/_buttons.scss]: There's an extra space here.

File: packages/spark-core/components/_buttons.scss [hunk 2]
```diff
@@ -109,3 +109,21 @@
...
+  background-color: $btn-secondary-disabled-background-color;
+  border-color: $btn-secondary-disabled-border-color;
+  color: $btn-secondary-disabled-text-color;
+}
+
+.sprk-c-Button--tertiary.sprk-is-Disabled,
+.sprk-c-Button--tertiary[disabled] {
+  box-shadow: none;
+  border: none;
+
```
Comment (afebbraro) [inline packages/spark-core/components/_buttons.scss]: There's an extra space here.

File: packages/spark-core/settings/_colors.scss [hunk 3]
```diff
@@ -4,33 +4,26 @@
 
 // Primary
 $red: rgb(200, 16, 46);
-$black: rgb(0, 0, 0);
+$black: rgb(51, 51, 51);
```
Comment (afebbraro) [inline packages/spark-core/settings/_colors.scss]: I saw color changes in another PR you did for Links. I'm guessing you needed it in both PRs?

File: packages/spark-core/settings/_settings.scss [hunk 4]
```diff
@@ -29,7 +29,7 @@ $space-inset-xl: 64px 64px 64px 64px !default;
 $space-inset-short-xs: 2px 4px 2px 4px !default;
 $space-inset-short-s: 4px 8px 4px 8px !default;
 $space-inset-short-m: 8px 16px 8px 16px !default;
-$space-inset-short-l: 16px 32px 16px 32px !default;
+$space-inset-short-l: 16px 24px 16px 24px !default;
```
Comment (afebbraro) [inline packages/spark-core/settings/_settings.scss]: Why was this change made?

File: packages/spark-core/settings/_settings.scss [hunk 5]
```diff
@@ -269,40 +269,46 @@ $promo-subtitle-font-weight: $font-weight-body-one !default;
 //
 // Buttons
 ///
-$btn-background-color: $red !default;
-$btn-shadow: 0 2px 2px rgba(0, 0, 0, 0.2) !default;
+$btn-background-color: $green !default;
+$btn-shadow: 0px 3px 10px rgba(20,20,20,0.08) !default;
```
Comment (afebbraro) [inline packages/spark-core/settings/_settings.scss]: Don't need the px after the 0.

File: packages/spark-core/settings/_settings.scss [hunk 6]
```diff
@@ -269,40 +269,46 @@ $promo-subtitle-font-weight: $font-weight-body-one !default;
...
 $btn-padding: $space-inset-short-l !default;
 $btn-breakpoint-xs: 30rem !default;
 $btn-breakpoint-s: 42.5rem !default;
 $btn-transition-speed: 0.3s !default;
 // Disabled
 $btn-disabled-opacity: 1 !default;
-$btn-disabled-background-color: $gray !default;
-$btn-disabled-text-color: $black-tint-50 !default;
 $btn-disabled-border-width: $btn-border-width !default;
+$btn-disabled-background-color: $gray-dark !default;//$gray !default;
```
Comment (afebbraro) [inline packages/spark-core/settings/_settings.scss]: There's some comment code here thats gotta be removed.

File: packages/spark-core/settings/_settings.scss [hunk 7]
```diff
@@ -269,40 +269,46 @@ $promo-subtitle-font-weight: $font-weight-body-one !default;
...
+$btn-disabled-text-color: $white !default;
 $btn-disabled-border-color: transparent !default;
+$btn-secondary-disabled-background-color: transparent !default;//$gray !default;
```
Comment (afebbraro) [inline packages/spark-core/settings/_settings.scss]: There's some comment code here thats gotta be removed.

File: packages/spark-core/settings/_settings.scss [hunk 8]
```diff
@@ -269,40 +269,46 @@ $promo-subtitle-font-weight: $font-weight-body-one !default;
...
-$btn-secondary-border-color: $black !default;
+$btn-secondary-background-color: $white !default;
+$btn-secondary-border-color: $gray-dark !default;
 $btn-secondary-text-color: $black !default;
-$btn-secondary-hover-text-color: $red !default;
+$btn-secondary-hover-text-color: $black !default;
 $btn-secondary-hover-background-color: transparent !default;
-$btn-secondary-hover-border-color: $red !default;
+$btn-secondary-hover-border-color: $gray-dark !default;
+$btn-secondary-shadow: 0px 3px 10px rgba(20,20,20,0.08) !default;
```
Comment (afebbraro) [inline packages/spark-core/settings/_settings.scss]: Don't need the px unit after the 0.

File: src/patterns/fundamentals/color/collection.yaml [hunk 9]
```diff
@@ -1,6 +1,5 @@
 description:
-  These are the colors that have been chosen to represent the brand. These colors should be used in the correct
-  proportion (see below) in all components of an application that require the use of color. The colors are set as
+  These are the colors that have been chosen to represent the brand. The colors are set as
   variables in Spark and as document colors in the Design Toolkit.
```
Comment (afebbraro) [inline src/patterns/fundamentals/color/collection.yaml]: I don't think the Design Toolkit has these colors yet.

File: src/patterns/utilities/single-property/color.hbs [hunk 10]
```diff
@@ -16,46 +16,47 @@ description: These utilities control CSS properties related to color and backgro
...
+      <td>Sets the font color to blue.</td>
     </tr>
     <tr>
       <td class="sprk-u-FontWeight--bold">.sprk-u-Color--gray</td>
       <td>Sets the font color to gray.</td>
     </tr>
     <tr>
-      <td class="sprk-u-FontWeight--bold">.sprk-u-Color--red-mid</td>
-      <td>Sets the font color to red-mid.</td>
+      <td class="sprk-u-FontWeight--bold">.sprk-u-Color--green</td>
```
Comment (afebbraro) [inline src/patterns/utilities/single-property/color.hbs]: What were these changes in here for? Looks like moving the color utilities around?

File: src/patterns/utilities/single-property/color.hbs [hunk 11]
```diff
@@ -73,44 +74,84 @@ description: These utilities control CSS properties related to color and backgro
...
     </tr>
     <tr>
-      <td class="sprk-u-FontWeight--bold">.sprk-u-BackgroundColor--green</td>
-      <td>Sets the background color to green.</td>
+      <td class="sprk-u-FontWeight--bold">.sprk-u-BackgroundColor--red</td>
+      <td>Sets the background color to red.</td>
+    </tr>
+    <tr>
+      <td class="sprk-u-FontWeight--bold">.sprk-u-BackgroundColor--red-tint-75</td>
+      <td>Sets the background color to red-tint-75.</td>
```
Comment (afebbraro) [inline src/patterns/utilities/single-property/color.hbs]: This description should be like the other ones, for example, " green tinted at 50%.

File: src/patterns/utilities/single-property/color.hbs [hunk 12]
```diff
@@ -73,44 +74,84 @@ description: These utilities control CSS properties related to color and backgro
...
+    </tr>
+    <tr>
+      <td class="sprk-u-FontWeight--bold">.sprk-u-BackgroundColor--red-tint-50</td>
+      <td>Sets the background color to red-tint-50.</td>
```
Comment (afebbraro) [inline src/patterns/utilities/single-property/color.hbs]: This description should be like the other ones, for example, " green tinted at 50%.

File: src/patterns/utilities/single-property/color.hbs [hunk 13]
```diff
@@ -73,44 +74,84 @@ description: These utilities control CSS properties related to color and backgro
...
+    </tr>
+    <tr>
+      <td class="sprk-u-FontWeight--bold">.sprk-u-BackgroundColor--red-tint-25</td>
+      <td>Sets the background color to red-tint-25.</td>
```
Comment (afebbraro) [inline src/patterns/utilities/single-property/color.hbs]: This description should be like the other ones, for example, " green tinted at 50%.

File: src/patterns/utilities/single-property/color.hbs [hunk 14]
```diff
@@ -73,44 +74,84 @@ description: These utilities control CSS properties related to color and backgro
...
     </tr>
     <tr>
       <td class="sprk-u-FontWeight--bold">.sprk-u-BackgroundColor--yellow</td>
       <td>Sets the background color to yellow.</td>
     </tr>
     <tr>
-      <td class="sprk-u-FontWeight--bold">.sprk-u-BackgroundColor--blue</td>
-      <td>Sets the background color to blue.</td>
+      <td class="sprk-u-FontWeight--bold">.sprk-u-BackgroundColor--yellow-tint-75</td>
+      <td>Sets the background color to yellow-tint-75.</td>
```
Comment (afebbraro) [inline src/patterns/utilities/single-property/color.hbs]: This description should be like the other ones, for example, " green tinted at 50%.

File: src/patterns/utilities/single-property/color.hbs [hunk 15]
```diff
@@ -73,44 +74,84 @@ description: These utilities control CSS properties related to color and backgro
...
+    </tr>
+    <tr>
+      <td class="sprk-u-FontWeight--bold">.sprk-u-BackgroundColor--yellow-tint-50</td>
+      <td>Sets the background color to yellow-tint-50.</td>
```
Comment (afebbraro) [inline src/patterns/utilities/single-property/color.hbs]: This description should be like the other ones, for example, " green tinted at 50%.

File: src/patterns/utilities/single-property/color.hbs [hunk 16]
```diff
@@ -73,44 +74,84 @@ description: These utilities control CSS properties related to color and backgro
...
+    </tr>
+    <tr>
+      <td class="sprk-u-FontWeight--bold">.sprk-u-BackgroundColor--yellow-tint-25</td>
+      <td>Sets the background color to yellow-tint-25.</td>
```
Comment (afebbraro) [inline src/patterns/utilities/single-property/color.hbs]: This description should be like the other ones, for example, " green tinted at 50%.

File: packages/spark-core/components/_buttons.scss [hunk 17]
```diff
@@ -60,9 +61,8 @@
 .sprk-c-Button--tertiary {
   background-color: $btn-tertiary-background-color;
   box-shadow: none;
-  border-color: $btn-tertiary-border-color;
+  border: none;
```
Comment (afebbraro) [inline packages/spark-core/components/_buttons.scss]: I think the variable should be left here but zeroed out in the _settings.scss file. You can create a new name to reflect the whole border property.

File: packages/spark-core/components/_buttons.scss [hunk 18]
```diff
@@ -80,12 +80,12 @@
   &:active::after,
   &:focus::after {
     border-color: $btn-tertiary-hover-line-color;
-    animation: width-zero-to-full $btn-transition-speed ease-in-out;
```
Comment (afebbraro) [inline packages/spark-core/components/_buttons.scss]: With this change, do we still need this animation code? Is it something that we can go and also remove?

File: packages/spark-core/components/_buttons.scss [hunk 19]
```diff
@@ -80,12 +80,12 @@
...
+    transition: border $btn-transition-speed ease-in-out;
   }
```
Comment (afebbraro) [inline packages/spark-core/components/_buttons.scss]: I don't think this transition is coming through. It needs to be set not in the hover, active ETC states, but in default state.

File: packages/spark-core/components/_buttons.scss [hunk 20]
```diff
@@ -109,3 +109,21 @@
     color: $btn-disabled-text-color;
   }
 }
+
+.sprk-c-Button--secondary.sprk-is-Disabled,
+.sprk-c-Button--secondary[disabled] {
+  box-shadow: none;
```
Comment (afebbraro) [inline packages/spark-core/components/_buttons.scss]: These box-shadows and borders need to be set as vars and zeroed out in the settings file.

File: packages/spark-core/components/_buttons.scss [hunk 21]
```diff
@@ -109,3 +109,21 @@
...
+
+  background-color: $btn-secondary-disabled-background-color;
+  border-color: $btn-secondary-disabled-border-color;
+  color: $btn-secondary-disabled-text-color;
+}
+
+.sprk-c-Button--tertiary.sprk-is-Disabled,
+.sprk-c-Button--tertiary[disabled] {
+  box-shadow: none;
```
Comment (afebbraro) [inline packages/spark-core/components/_buttons.scss]: These box-shadows and borders need to be set as vars and zeroed out in the settings file.

File: packages/spark-core/settings/_settings.scss (hunk 4 above)
Comment (bobwise) [reply packages/spark-core/settings/_settings.scss]: Button padding was changed in the new design doc.

File: src/patterns/utilities/single-property/color.hbs (hunk 10 above)
Comment (bobwise) [reply src/patterns/utilities/single-property/color.hbs]: I reordered the colors to match the order in the design doc.

**Comment by bobwise** [comment]
> Is the secondary supposed to be used next to a Primary button? Ex. Highlight board and the secondary button's "lifting" appearance on hover next to the primary which does not "lift" up on hover.

Primary and Secondary buttons have the same box shadow applied on hover. It is a subtle shadow and it is more noticeable on the secondary button. I can barely see it on my monitor but the Inspector assures me it is there.

>The other question is about the secondary button's active state, I noticed it isnt the same as the hover. Is that meant to be?

I will make this change.

File: packages/spark-core/components/_buttons.scss (hunk 18 above)
Comment (bobwise) [reply packages/spark-core/components/_buttons.scss]: width-zero-to-full is still used in a couple places.

File: packages/spark-core/components/_buttons.scss (hunk 19 above)
Comment (bobwise) [reply packages/spark-core/components/_buttons.scss]: The desired behavior is that the underline on tertiary buttons fades from $gray-dark to $black over .3 seconds when hovered.

File: packages/spark-core/settings/_settings.scss (hunk 4 above)
Comment (afebbraro) [reply packages/spark-core/settings/_settings.scss]: We don't want to change this value. This is on the normal scale. You'll want to create a var for buttons. Something like $button-padding and set that to $space-m $sprk-space-misc-a; The misc spacing stuff is in my PR though lol.

File: src/patterns/utilities/single-property/color.hbs (hunk 11 above)
Comment (afebbraro) [reply src/patterns/utilities/single-property/color.hbs]: Forgot this one ^

File: src/patterns/utilities/single-property/color.hbs (hunk 12 above)
Comment (afebbraro) [reply src/patterns/utilities/single-property/color.hbs]: Forgot this one ^

File: packages/spark-core/settings/_settings.scss [hunk 22]
```diff
@@ -314,7 +328,7 @@ $link-transition: 0.3s !default;
 $link-underline-color: $red !default;
 $link-underline-visited-color: $red !default;
 $link-underline-width: 0.0625em !default;
-$link-visited-color: $red-mid !default;
+$link-visited-color: $red !default;
```
Comment (afebbraro) [inline packages/spark-core/settings/_settings.scss]: You'll need to update the link page Sass Table to reflect the new default.

File: packages/spark-core/settings/_settings.scss [hunk 23]
```diff
@@ -466,7 +480,7 @@ $dropdown-padding: $space-inset-short-l !default;
 $dropdown-header-padding: $space-inset-short-l !default;
 $dropdown-footer-padding: $space-inset-short-l !default;
 $dropdown-border: 1px solid $gray !default;
-$dropdown-active-background-color: $gray-tint-75 !default;
+$dropdown-active-background-color: $gray !default;
```
Comment (afebbraro) [inline packages/spark-core/settings/_settings.scss]: Make sure you update the dropdown page Sass Var Table to reflect the new default color.

File: packages/spark-core/settings/_settings.scss [hunk 24]
```diff
@@ -495,7 +509,7 @@ $masthead-border-bottom: $masthead-border-bottom-size solid $gray !default;
 $masthead-breakpoint: 54rem !default;
 $masthead-column-width: 89rem !default;
 $big-nav-column-width: 64rem !default;
-$big-nav-bg: $gray-tint-25 !default;
+$big-nav-bg: $gray !default;
```
Comment (afebbraro) [inline packages/spark-core/settings/_settings.scss]: Make sure you update the masthead page Sass Table to reflect the new default big nav bg color.

File: packages/spark-core/settings/_settings.scss (hunk 4 above)
Comment (afebbraro) [reply packages/spark-core/settings/_settings.scss]: @yodasw16  this affects everything that is using that variable. So, we shouldn't move forward with this.

Comment (afebbraro) [reply packages/spark-core/settings/_settings.scss]: What we can do in the meantime is this `$button-padding: $space-m 24px`

**Comment by bbrown7** [comment]
Secondary button border should also change to #333 on hover

**Comment by yodasw16** [comment]
Replaced with #958

**Pull request closed without merging**